### ISSUE TITLE
dBFT stage 2: BLS

### ISSUE THREAD

**Issue by roman-khimov** [bane-labs/go-ethereum#12]
Unfortunately, we can't make it work properly for multiple signers with sec256r1 or sec256k1. The reason is that unlike Neo blocks where we have invocation/verification scripts that are _not_ a part of the hashed data, Ethereum blocks are hashed as a whole. This leads to an interesting signature-generation problem (signed hash is not the same as the resulting block hash), but more importantly it also means that whatever signature we fit into the block must be stable. Neo blocks can have different sets of signatures collected by different CNs attached to the same block and any set works fine, Ethereum has to have a single proper signature.

Luckily we can have it with BLS and once upon a time this was already done to NeoGo as an experiment: https://github.com/nspcc-dev/neo-go/compare/master...bls

It's not that hard, dBFT library interface itself allows for that to happen quite easily. So change the keys and make us check BLS. 

Depends on #11.

**Comment by doubiliu** [comment]
I don't have much objection to the proposal to replace it with threshold signatures. But this thing has a more important thing (distribution and confirmation of shared private keys). If we use threshold signatures, we must build an additional website or introduce an automated shared private key distribution verification process in DBFT. If you do not perform this operation, it is very easy to cause error.

**Comment by AnnaShaleva** [comment]
As a part of this task we would probably need our own MIME type to abstract private key, ref. `https://github.com/bane-labs/go-ethereum/pull/28#discussion_r1395580777.

**Comment by AnnaShaleva** [comment]
I'd propose to close this for now, as BLS isn't suitable for our needs because of non-constant signature (and, consequently, non-constant block hash). The solution is to go with an existing signing scheme, to extend it for proper multisignature check (ref. #40) and rely on native fork resolving mechanism.

@roman-khimov, close?

**Comment by roman-khimov** [comment]
Replaced with #48.
